hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/StreamToStreamJoinPhysicalRule.java

```diff
@@ -93,6 +93,13 @@ public void onMatch(RelOptRuleCall call) {
                 metadataQuery(left).extractWatermarkedFields(left),
                 metadataQuery(right).extractWatermarkedFields(right));
 
+        if (wmFields == null) {
```

Comment:
I think watermark field presence check would be more readable if done in `onMatch` before `watermarkedFields` invocation and `watermarkedFields` would require non-null arguments.

I believe that there is no case when both joined streams have watermark but cannot be joined because the watermarks are "not compatible".

Reply:
@k-jamroz To me, both approaches have similar readability, I wouldn't change anything. But if you have strong intentions, I may change it.

Reply:
For me passing nulls to get null from `watermarkedFields` so we know that nulls (i.e. streams without watermarks) are not supported seems convoluted. I strongly prefer leftFields and rightFields null check in `onMatch`.